mqt/qmap/subarchitectures.py

```diff
@@ -0,0 +1,275 @@
+"""
+Functionality for computing good subarchitectures for quantum circuit mapping.
+
+The function provided in this module... TODO
+"""
+from __future__ import annotations
+
+import os
+import pickle
+from collections import defaultdict
+from itertools import combinations
+from typing import Dict, List, NewType, Tuple, Union
+
+import networkx as nx
+import retworkx as rx
+
+Subarchitecture = NewType("Subarchitecture", Union[rx.PyGraph, List[Tuple[int, int]]])
+PartialOrder = NewType("PartialOrder", Dict[Tuple[int, int], Tuple[int, int]])
```

Comment:
Is there? mypy doesn't suggest me anything better.

Reply:
Ah, maybe because of the outer `NewType` it's not possible to write something like:
`NewType("Subarchitecture", rx.PyGraph | list[tuple[int, int]]])`
My bad. feel free to resolve this.